storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -444,13 +445,30 @@ func (d *DotGit) rewritePackedRefsWithoutRef(name plumbing.ReferenceName) (err e
 
 		return err
 	}
+	doCloseF := true
```

Comment:
Would it be cleaner to call `ioutil.CheckClose` with a `sync.Once`?

Reply:
I don't know!  That seems a little heavy weight since it involves mutexes/atomicity, which isn't needed here in a single-flow function.  For that reason I personally prefer this way, but if you feel strongly about it let me know.

Reply:
Sounds good; I'm OK with this way. Thanks!